subprojects/platform-native/src/docs/sample-templates/source-globbing/README.adoc

```diff
@@ -0,0 +1,165 @@
+:c: C
+:source-file-extension: c
+:header-file-extension: h
+:application: library
+endif::[]
+
+ifeval::["{plugin-name}" == "cpp-library"]
+:c-application: {cpp} library
+:an-application: a library
+:c: {cpp}
+:source-file-extension: cpp
+:header-file-extension: hpp
+:application: library
+endif::[]
+
+ifeval::["{plugin-name}" == "objective-c-library"]
+:c-application: Objective-C library
+:an-application: a library
+:c: Objective-C
+:source-file-extension: m
+:header-file-extension: h
+:application: library
+endif::[]
+
+ifeval::["{plugin-name}" == "objective-cpp-library"]
+:c-application: Objective-{cpp} library
+:an-application: a library
+:c: Objective-{cpp}
+:source-file-extension: mm
+:header-file-extension: hpp
+:application: library
+endif::[]
+
+ifeval::["{plugin-name}" == "swift-library"]
+:c-application: Swift library
+:an-application: a library
+:c: Swift
+:source-file-extension: swift
+:application: library
+endif::[]
+
+:ref-plugin: <<plugin:{plugin-name}, {c-application} plugin>>
+
+ifndef::c[ERROR: Please define `plugin-name` attribute.]
+
+= {c-application} Custom Source Layout Globbing
+:summary: Set custom source layout Globbing for a {c-application}.
+:type: sample-chapter
+:tags: sample, {application}, sources, native, {C}, gradle
+:category: {C}
+:description: See how to set the source layout globbing of {an-application} implemented in {C} using the Gradle Nokee plugins.
+:ref-configurable-file-tree: https://docs.gradle.org/current/javadoc/org/gradle/api/file/ConfigurableFileTree.html
+:ref-configurable-file-tree-exclude: https://docs.gradle.org/current/javadoc/org/gradle/api/tasks/util/PatternFilterable.html#exclude-java.lang.String...-
+:ref-configurable-file-tree-matching: https://docs.gradle.org/current/javadoc/org/gradle/api/file/FileTree.html#matching-groovy.lang.Closure-
+
+NOTE: You can open this sample inside an IDE using the https://www.jetbrains.com/help/idea/gradle.html#gradle_import_project_start[IntelliJ native importer] or https://projects.eclipse.org/projects/tools.buildship[Eclipse Buildship].
+
+Here is a sample where we set the source layout globbing of {an-application} implemented in {C} with no dependencies and minimal configuration.
+
+====
+[.multi-language-sample]
+=====
+.build.gradle
+[source,groovy]
+----
+include::groovy-dsl/build.{plugin-name}.gradle[]
+----
+=====
+[.multi-language-sample]
+=====
+.build.gradle.kts
+[source,kotlin]
+----
+include::kotlin-dsl/build.{plugin-name}.gradle.kts[]
+----
+=====
+====
+<1> Apply the {ref-plugin}.
+<2> Set {C} sources to *include* all files with `{source-file-extension}` extension recursively from the `srcs` directory.
+ifdef::header-file-extension[]
+<3> Set private headers to *include* all files with `{header-file-extension}` extension recursively from the `hdrs` directory.
+ifeval::["{entry-point}" == "library"]
+<4> Set public headers to *include* all files with `{header-file-extension}` extension recursively from the `incs` directory.
```

Comment:
❓ I'm really unsure about showing private/public headers configuration because we will need to note something a bit more complicated. Headers do not work like the source files, meaning you can include a broken header, but as long as you don't `#include` the file, nothing bad will happen. As mentioned previously, the compilation only requires the including root (aka header search paths). In the case of a `fileTree`, only the base directory is really necessary. However, the _glob_ pattern is useful to know which files should be considered valid headers. There are current issues where that pattern will be super useful, however, we don't use the pattern yet. I still want users to start using those patterns where it makes sense regardless if the feature is present or not in Nokee. I'm just unsure how to "properly" explain it here.

I think the important parts are:
- `fileTree` base directory is used as header search paths during compilation
- glob patterns are used for header isolation (we could point to what header isolation is but I don't think it's important for this sample)